homeassistant/components/device_tracker/asuswrt.py

```diff
@@ -208,7 +208,8 @@ def _get_leases(self):
             if host == '*':
                 host = ''
             mac = device['mac'].upper()
-            devices[mac] = Device(mac, device['ip'], host)
+            if mac in cur_devices.keys():
```

Comment:
Why do you limit this to devices already reported so far? Sure, everything that got a lease from us should also be in the ARP neighborhood, but that would make this check superfluous. If the router knows about leases for devices that are not in the ARP table, why not also report them?

Reply:
We must combine the information from both leases and neighborhood to get all the information we need; MAC, host name, and current status. 
MAC from leases
Host name from leases
status from neighborhood

the value that we can use to merge the tables is the IP address.

Reply:
Remove the `.keys()`. Checking membership of keys in dict works without specifically asking for the keys.